js_modules/dagit/packages/core/src/app/__tests__/AppTopNav.test.tsx

```diff
@@ -75,17 +75,16 @@ describe('AppTopNav', () => {
       </TestProvider>,
     );
 
-    await waitFor(() => {
-      const runsLink = screen.getByRole('link', {name: /runs/i});
-      expect(runsLink.closest('a')).toHaveAttribute('href', '/runs');
-      expect(screen.getByText('Assets').closest('a')).toHaveAttribute('href', '/assets');
-      expect(screen.getByText('Deployment').closest('a')).toHaveAttribute('href', '/locations');
-      expect(screen.getByText('RightOfSearchBar')).toBeVisible();
-    });
+    const runsLink = await screen.findByRole('link', {name: /runs/i});
+    expect(runsLink.closest('a')).toHaveAttribute('href', '/runs');
+    expect(screen.getByText('Assets').closest('a')).toHaveAttribute('href', '/assets');
+    expect(screen.getByText('Deployment').closest('a')).toHaveAttribute('href', '/locations');
+    expect(screen.getByText('RightOfSearchBar')).toBeVisible();
   });
 
   describe('Repo location errors', () => {
     it('does not show warning icon when no errors', async () => {
+      // Act immediately since we're asserting a post-query null state.
```

Comment:
why not `waitFor` on the line below? is it too early?

Reply:
My thought here is that it's not really correct to "wait for" the null state, since it might be null immediately *and* after the query is complete.